metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/configuration/MetadataImporterMainApplicationConfiguration.java

```diff
@@ -47,34 +51,59 @@
+                sraAnalysisXmlParser(),
+                analysisConverter(),
+                fileExtractorFromAnalysis(fileRepository),
+                sraAssemblyXmlParser(),
+                referenceSequenceConverter(),
+                analysisRepository,
+                studyRepository,
+                referenceSequenceRepository
+        );
     }
 
     @Bean
     @ConditionalOnProperty(name = "import.source", havingValue = "DB")
-    public ObjectsImporter objectImporterThroughEnaDatabase(
-            SraXmlRetrieverThroughDatabase sraXmlRetrieverThroughDatabase,
-            PublicationRepository publicationRepository,
-            WebResourceRepository webResourceRepository,
-            FileRepository fileRepository,
-            TaxonomyRepository taxonomyRepository,
-            AnalysisRepository analysisRepository,
-            StudyRepository studyRepository) {
-        return new SraObjectsImporterThroughDatabase(sraXmlRetrieverThroughDatabase, sraStudyXmlParser(),
-                studyConverter(), publicationExtractorFromStudy(publicationRepository),
-                webResourceExtractorFromStudy(webResourceRepository), taxonomyExtractor(taxonomyRepository),
-                sraAnalysisXmlParser(), analysisConverter(), fileExtractorFromAnalysis(fileRepository),
-                analysisRepository,studyRepository);
+    public ObjectsImporter objectImporterThroughEnaDatabase(SraXmlRetrieverThroughDatabase sraXmlRetrieverThroughDatabase,
+                                                PublicationRepository publicationRepository,
+                                                WebResourceRepository webResourceRepository,
+                                                FileRepository fileRepository,
+                                                TaxonomyRepository taxonomyRepository,
+                                                ReferenceSequenceRepository referenceSequenceRepository,
+                                                AnalysisRepository analysisRepository,
+                                                StudyRepository studyRepository) {
+        return new SraObjectsImporterThroughDatabase(
+                sraXmlRetrieverThroughDatabase,
+                sraStudyXmlParser(),
+                studyConverter(),
+                publicationExtractorFromStudy(publicationRepository),
+                webResourceExtractorFromStudy(webResourceRepository),
+                taxonomyExtractor(taxonomyRepository),
+                taxonomyExtractorFromReferenceSequence(taxonomyRepository),
+                sraAnalysisXmlParser(),
+                analysisConverter(),
+                fileExtractorFromAnalysis(fileRepository),
+                sraAssemblyXmlParser(),
+                referenceSequenceConverter(),
+                analysisRepository,
+                studyRepository,
+                referenceSequenceRepository
+        );
     }
 
     private StudyConverter studyConverter() {
```

Comment:
I noticed this problem as well and have already performed a conceptual grouping in #55. I suggest that in this PR we leave this as is to avoid unnecessary downstream rebasing.